frontend/apps/reader/modules/readerlink.lua

```diff
@@ -576,56 +628,60 @@ function ReaderLink:onGotoLink(link, neglect_current_location, allow_footnote_po
+                {
+                    {
+                        text = _("Cancel"),
+                        callback = function()
+                            UIManager:close(choose_action)
+                        end,
+                    },
+                    {
+                        text = _("Add to Wallabag"),
+                        callback = function()
+                            self.ui.wallabag:addArticle(link_url)
+                            UIManager:close(choose_action)
+                        end,
+                    },
 
-    -- Check if it is a link to a local file
-    local linked_filename = link_url:gsub("^file:", "") -- remove local file protocol if any
-    local anchor
-    if linked_filename:find("?") then -- remove any query string (including any following anchor)
-        linked_filename, anchor = linked_filename:match("^(.-)(%?.*)$")
-    elseif linked_filename:find("#") then -- remove any anchor
-        linked_filename, anchor = linked_filename:match("^(.-)(#.*)$")
-    end
-    linked_filename  = ffiutil.joinPath(self.document_dir, linked_filename) -- get full path
-    linked_filename = ffiutil.realpath(linked_filename) -- clean full path from ./ or ../
-    if linked_filename and lfs.attributes(linked_filename, "mode") == "file" then
-        local DocumentRegistry = require("document/documentregistry")
-        local provider = DocumentRegistry:getProvider(linked_filename)
-        if provider then
-            -- Display filename with anchor or query string, so the user gets
-            -- this information and can manually go to the appropriate place
-            local display_filename = linked_filename
-            if anchor then
-                display_filename = display_filename .. anchor
+                },
+            }
+            if Device:openLink() == nil then
+                table.insert(buttons, {
+                    {
+                        text = "–",
+                        enabled = false,
+                    },
+                    {
+                        text = _("Open in browser"),
+                        callback = function()
+                            Device:openLink(link_url)
+                            UIManager:close(choose_action)
+                        end,
+                    },
+                })
             end
-            UIManager:show(ConfirmBox:new{
-                text = T(_("Would you like to read this local document?\n\n%1\n"), display_filename),
-                ok_callback = function()
-                    UIManager:scheduleIn(0.1, function()
-                        self.ui:switchDocument(linked_filename)
-                    end)
-                end
-            })
-        else
-            UIManager:show(InfoMessage:new{
-                text = T(_("Link to unsupported local file:\n%1"), link_url),
-            })
+
+            choose_action = ButtonDialogTitle:new{
+                title = T(_("External link:\n\n'%1'"), link_url),
```

Comment:
Not fond of the url inside `' '` on the screenshots. Can't you just let it plain?